### DISCUSSION TITLE
Why is there no Dropdown / Select with self-managed open state?

### DISCUSSION THREAD

**Discussion by jotak** [patternfly/patternfly#5473] [Feature requests]
All the `Dropdown`, `Select` and their variations seem to require their consumer to manage the "open" state, it seems like a curious choice to me, as I would imagine it's the nominal use case to have these components on "open" state only when the user clicks on it, and closed otherwise.

In my team we use a bunch of these components, and everytime have to manage the open state, one per Dropdown, in the parent. I'm thinking about writing my own generic intermediate, with something like:

```typescript
const SimplerSelect = (props: Omit<SelectProps, 'isExpanded'>) => {
  const [isExpanded, setExpanded] = React.useState(false);
  return (
    <Select
      {...props}
      onToggle={isExpanded => {
        if (props.onToggle) {
          props.onToggle(isExpanded);
        }
        setExpanded(isExpanded);
      }}
      isExpanded={isExpanded}
    >
      {props.children}
    </Select>
  );
};
```

but I wonder why this wouldn't exist directly in PF

**Comment by jotak** [comment]
anyone here?

**Comment by Andrewgdewar** [comment]
Ditto, due to this, I pretty much always make a wrapper to contain state and use that instead of the base component. 
[Example](https://github.com/content-services/content-sources-frontend/blob/main/src/components/DropdownSelect/DropdownSelect.tsx)

**Comment by nicolethoen** [comment]
This is an old thread - i'm sorry.
I will note that generally, it's been standard practice for PatternFly to defer as much business logic as we can to the consumer. So all of our menu like components tend to not be controlled in the way you describe. Generally we don't have much push back, but when we do, we can sometimes implement business logic or create a new controlled variant. Feel free to open an issue if you'd like to kick off that conversation.
